dali/python/nvidia/dali/pipeline.py

```diff
@@ -131,24 +150,38 @@ def __init__(self, batch_size = -1, num_threads = -1, device_id = -1, seed = -1,
         self._device_id = device_id
         self._seed = seed if seed is not None else -1
         self._exec_pipelined = exec_pipelined
+        # When initializing DALI, we do the following in order:
+        # * Discover the ops specified in Python, group the ExternalSources (_build_graph())
+        # * Start the Python workers pool (_start_py_workers())
+        # * Construct the C++ Pipeline backend and pass the graph to it (_init_pipeline_backend())
+        # * Build the pieline. (_pipe.Build())
+        self._py_graph_built = False
+        self._py_pool_started = False
+        self._backend_prepared = False
```

Comment:
Maybe instead of having a plethora of variables and every time checking all of them we should have an entity that would track the state when queried?

Reply:
I tried to group them in one place. Do we want some kind of state machine for that? Maybe good idea for a followup.

Reply:
I tried to group them in one place. Do we want some kind of state machine for that? Maybe good idea for a followup.

Reply:
Yep, follow up it is.